navigator/frontend/src/ui-core/src/api/DamlLfValue.ts

```diff
@@ -174,13 +205,15 @@ export function toJSON(value: DamlLfValue): JSON {
     case 'party':       return value.value;
     case 'unit':        return {};
     case 'optional':    return value.value === null ? null : toJSON(value.value);
-    case 'list':        return value.value.map((e) => toJSON(e))
+    case 'list':        return value.value.map((e) => toJSON(e));
     case 'record':
       const r: {[label: string]: JSON} = {};
       value.fields.forEach((f) => r[f.label] = toJSON(f.value));
       return r;
     case 'variant':
-      return {[value.constructor]: toJSON(value.value)}
+      return {[value.constructor]: toJSON(value.value)};
+    case 'map':
+      return value.value.map((e) => ({key: e.key, value: toJSON(e.value)}));
```

Comment:
Similar to other places, I would make this an object with many keys, instead of an array of key-value pairs. This function is mainly used in `frontend-config.js` (scripting of the UI), and `toJSON(argument).foo.bar` is easier to write than `toJSON(argument).foo.find(f => f.key === 'bar')`